pyteal/ast/subroutine_test.py

```diff
@@ -40,6 +44,15 @@ def fnWithOnlyArgExprAnnotations(a: Expr, b: Expr):
     def fnWithPartialExprAnnotations(a, b: Expr) -> Expr:
         return Return()
 
+    def fnWithReturnTernaryExprAnnotations(a, b: Expr) -> Nonce:
```

Comment:
@jasonpaulos @ahangsu 

I have a PR out that makes the following proposal - add `input_types` optional paramater to **Subroutine** decorator. Will this contradict the approach here, or can they work together? EG:

```python
@Subroutine(TealType.bytes, input_types=[TealType.bytes, TealType.uint64])
def string_mult(s: ScratchVar, n):
    i = ScratchVar(TealType.uint64)
    tmp = ScratchVar(TealType.bytes)
    start = Seq(i.store(Int(1)), tmp.store(s.load()), s.store(Bytes("")))
    step = Seq(i.store(i.load() + 1), s.store(Concat(s.load(), tmp.load())))
    return Seq(
        For(start, i.load() <= n, step),
        s.load(),
    )
```